### PR TITLE
feat(query): added Network Interfaces With Public IP for Terraform

### PR DESCRIPTION
**Pull request by tiagocruz20**
**Proposed Changes**
- added a query to check if there is any public IP in Network Interfaces

I submit this contribution under the Apache-2.0 license.

### CODE REVIEW

File: assets/queries/terraform/azure/network_interfaces_dont_use_public_IPs/query.rego [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+CxPolicy[result] {
+	network := input.document[i].resource.azurerm_network_interface[name].ip_configuration
+
+	common_lib.valid_key(network, "public_ip_address_id")
+
+	result := {
+		"documentId": input.document[i].id,
+		"searchKey": sprintf("azurerm_network_interface[%s].ip_configuration.public_ip_address_id", [name]),
+		"issueType": "MissingAttribute",
```
Comment (felipe-avelar) [inline assets/queries/terraform/azure/network_interfaces_dont_use_public_IPs/query.rego:13]:
```suggestion
		"issueType": "IncorrectValue",
```

File: assets/queries/terraform/azure/network_interfaces_dont_use_public_IPs/query.rego [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+import data.generic.common as common_lib
+
+CxPolicy[result] {
+	network := input.document[i].resource.azurerm_network_interface[name].ip_configuration
+
+	common_lib.valid_key(network, "public_ip_address_id")
+
+	result := {
+		"documentId": input.document[i].id,
+		"searchKey": sprintf("azurerm_network_interface[%s].ip_configuration.public_ip_address_id", [name]),
```
Comment (felipe-avelar) [inline assets/queries/terraform/azure/network_interfaces_dont_use_public_IPs/query.rego:12]:
```suggestion
		"searchKey": sprintf("azurerm_network_interface.%s.ip_configuration.public_ip_address_id", [name]),
```

**Review by felipe-avelar — CHANGES_REQUESTED** [review]
Please, check comments

File: assets/queries/terraform/azure/network_interfaces_dont_use_public_IPs/query.rego (hunk 2 above)
Comment (rogeriopeixotocx) [reply assets/queries/terraform/azure/network_interfaces_dont_use_public_IPs/query.rego:12]: I think you should keep the [] in this case

Comment (felipe-avelar) [reply assets/queries/terraform/azure/network_interfaces_dont_use_public_IPs/query.rego:12]: Yes, It was a misleading by my side, this suggestion should be ignored

File: assets/queries/terraform/azure/network_interfaces_dont_use_public_IPs/metadata.json [hunk 3]
```diff
@@ -0,0 +1,11 @@
+{
+  "id": "c1573577-e494-4417-8854-7e119368dc8b",
+  "queryName": "Network Interfaces Don't Use Public IPs",
```
Comment (rogeriopeixotocx) [inline assets/queries/terraform/azure/network_interfaces_dont_use_public_IPs/metadata.json:3]: change directory name to `network_interface_with_public_ip`

```suggestion
  "queryName": "Network Interface With Public IP",
```

File: assets/queries/terraform/azure/network_interfaces_dont_use_public_IPs/metadata.json [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+  "severity": "MEDIUM",
+  "category": "Networking and Firewall",
+  "descriptionText": "Network Interfaces must not use public IPs",
```
Comment (rogeriopeixotocx) [inline assets/queries/terraform/azure/network_interfaces_dont_use_public_IPs/metadata.json:6]:
```suggestion
  "descriptionText": "Network Interfaces must not be exposed with a public IP address",
```

File: assets/queries/terraform/azure/network_interfaces_dont_use_public_IPs/test/positive_expected_result.json [hunk 5]
```diff
@@ -0,0 +1,8 @@
+[
+  {
+    "queryName": "Network Interfaces don't use public IPs",
```
Comment (rogeriopeixotocx) [inline assets/queries/terraform/azure/network_interfaces_dont_use_public_IPs/test/positive_expected_result.json:3]:
```suggestion
    "queryName": "Network Interface With Public IP",
```

**Review by rogeriopeixotocx — CHANGES_REQUESTED** [review]

**Review by rogeriopeixotocx — CHANGES_REQUESTED** [review]
you forgot to change the directory name to `network_interface_with_public_ip`

File: assets/queries/terraform/azure/network_interfaces_with_public_IP/query.rego [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+	network := input.document[i].resource.azurerm_network_interface[name].ip_configuration
+
+	common_lib.valid_key(network, "public_ip_address_id")
+
+	result := {
+		"documentId": input.document[i].id,
+		"searchKey": sprintf("azurerm_network_interface[%s].ip_configuration.public_ip_address_id", [name]),
+		"issueType": "IncorrectValue",
+		"keyExpectedValue": sprintf("'azurerm_network_interface[%s].ip_configuration.public_ip_address_id' is defined and not null", [name]),
+		"keyActualValue": sprintf("'azurerm_network_interface[%s].ip_configuration.public_ip_address_id' is undefined or null", [name]),
```
Comment (rafaela-soares) [inline assets/queries/terraform/azure/network_interfaces_with_public_IP/query.rego:15]:
```
"keyExpectedValue": sprintf("'azurerm_network_interface[%s].ip_configuration.public_ip_address_id' is undefined", [name]),
"keyActualValue": sprintf("'azurerm_network_interface[%s].ip_configuration.public_ip_address_id' is defined", [name]),
```

**Review by rafaela-soares — CHANGES_REQUESTED** [review]

**Review by rafaela-soares — DISMISSED** [review]
LGTM

**Review by felipe-avelar — APPROVED** [review]
LGTM

**Review by rogeriopeixotocx — CHANGES_REQUESTED** [review]
directory name should be always lowercase

**Review by rogeriopeixotocx — APPROVED** [review]

**Pull request merged**